android/java/org/chromium/chrome/browser/crypto_wallet/adapters/MarketCoinAdapter.java

```diff
@@ -0,0 +1,108 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.crypto_wallet.adapters;
+
+import static org.chromium.ui.base.ViewUtils.dpToPx;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+
+import org.chromium.brave_wallet.mojom.CoinMarket;
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.app.domain.MarketModel;
+import org.chromium.chrome.browser.app.helpers.ImageLoader;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class MarketCoinAdapter extends RecyclerView.Adapter<MarketCoinAdapter.ViewHolder> {
+    private static final int PERCENTAGE_CHANGE_DRAWABLE_PADDING_PX = 8;
+    private final MarketModel mMarketModel;
+    private final List<CoinMarket> mCoinMarkets;
+
+    private final Context mContext;
+
+    public MarketCoinAdapter(
+            @NonNull final Context context, @NonNull final MarketModel marketModel) {
+        mContext = context;
+        mMarketModel = marketModel;
+        mCoinMarkets = new ArrayList<>();
```

Comment:
nit: `mCoinMarkets = Collections.emptyList()`

Reply:
We can't change that. The empy list produced would be immutable, and it would produce a crash when calling `mCoinMarkets.add/addAll`.